File: sample-rest-api/src/test/java/nl/haploid/octowight/AbstractIT.java
package nl.haploid.octowight;

import nl.haploid.octowight.configuration.TestConfiguration;
import org.springframework.test.context.ContextConfiguration;
import org.springframework.test.context.junit4.AbstractTransactionalJUnit4SpringContextTests;
import org.springframework.test.context.support.AnnotationConfigContextLoader;

@ContextConfiguration(
		classes = {TestConfiguration.class},
		loader = AnnotationConfigContextLoader.class)
// TODO: @ComponentScan on TestConfiguration causes it to find all other @Configuration classes too =(
public abstract class AbstractIT extends AbstractTransactionalJUnit4SpringContextTests {
}

File: sample-resource-builder/src/test/scala/nl/haploid/octowight/sample/AbstractIT.scala
package nl.haploid.octowight.sample

import nl.haploid.octowight.sample.configuration.{TestConfiguration, KafkaConfiguration, PostgresConfiguration}
import org.scalatest._
import org.scalatest.tags.Slow
import org.springframework.test.context.support.AnnotationConfigContextLoader
import org.springframework.test.context.{ContextConfiguration, TestContextManager}

@ContextConfiguration(classes = Array(classOf[TestConfiguration], classOf[PostgresConfiguration], classOf[KafkaConfiguration]),
  loader = classOf[AnnotationConfigContextLoader])
@Slow
abstract class AbstractIT extends FlatSpec with BeforeAndAfterEach with ShouldMatchers {

  override def beforeEach() = new TestContextManager(this.getClass).prepareTestInstance(this)
}
